cleverhans/future/torch/attacks/deepfool.py

```diff
@@ -0,0 +1,86 @@
+"""
+DeepFool Attack from https://arxiv.org/pdf/1511.04599.pdf 
+by Seyed Mohsen et al.
+"""
+
+import torch
+import torch.nn as nn
+import numpy as np
+import copy
+
+from torch.autograd.gradcheck import zero_gradients as zgrad
+
+
+__all__ = ['deepfool']
+
+
+def deepfool(x, net:nn.Module, eta:float=0.02, tnc:int=10, miter:int=50, 
```

Comment:
For the other attacks we thought about specifying types but then decided to remain pythonic and leave it to the user to decide whether they want to pass a float or a tensor containing `eta` for instance.